patches/common/chromium/chore_add_debounce_on_the_updatewebcontentsvisibility_method_to.patch

```diff
@@ -0,0 +1,65 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Samuel Attard <sattard@slack-corp.com>
+Date: Wed, 5 Jun 2019 15:11:00 -0700
+Subject: chore: add debounce on the updateWebContentsVisibility method to
+ ensure quick changes in occlusion do not result in flickering
+
+diff --git a/content/app_shim_remote_cocoa/web_contents_view_cocoa.h b/content/app_shim_remote_cocoa/web_contents_view_cocoa.h
+index 230e259f6017310e556d11dec43973b74015f191..ebd19fc31efef50677da417dfd69f0c8cce6c682 100644
+--- a/content/app_shim_remote_cocoa/web_contents_view_cocoa.h
++++ b/content/app_shim_remote_cocoa/web_contents_view_cocoa.h
+@@ -58,6 +58,8 @@ CONTENT_EXPORT
+                        offset:(NSPoint)offset;
+ - (void)clearViewsHostableView;
+ - (void)updateWebContentsVisibility;
++- (remote_cocoa::mojom::Visibility)currentVisibility;
++- (void)notifyWebContentsVisibilityChanged;
+ - (void)viewDidBecomeFirstResponder:(NSNotification*)notification;
+ @end
+ 
+diff --git a/content/app_shim_remote_cocoa/web_contents_view_cocoa.mm b/content/app_shim_remote_cocoa/web_contents_view_cocoa.mm
+index 615fe671d415747cb84f673327629d3dc771039e..6c31ba5b9149161784f5813598ddcf30e2d8af2a 100644
+--- a/content/app_shim_remote_cocoa/web_contents_view_cocoa.mm
++++ b/content/app_shim_remote_cocoa/web_contents_view_cocoa.mm
+@@ -257,9 +257,14 @@ - (void)viewDidBecomeFirstResponder:(NSNotification*)notification {
+   host_->OnBecameFirstResponder(direction);
+ }
+ 
+-- (void)updateWebContentsVisibility {
++- (void)notifyWebContentsVisibilityChanged {
+   if (!host_)
+     return;
++
++  host_->OnWindowVisibilityChanged([self currentVisibility]);
++}
++
++- (Visibility)currentVisibility {
+   Visibility visibility = Visibility::kVisible;
+   if ([self isHiddenOrHasHiddenAncestor] || ![self window])
+     visibility = Visibility::kHidden;
+@@ -267,7 +272,24 @@ - (void)updateWebContentsVisibility {
+     visibility = Visibility::kVisible;
+   else
+     visibility = Visibility::kOccluded;
+-  host_->OnWindowVisibilityChanged(visibility);
++  return visibility;
++}
++
++- (void)updateWebContentsVisibility {
++  if (!host_)
```

Comment:
Is there a chance this could have unintended side effects for occlusion events we _do_ need to be notified of at the moment they occur?

Reply:
Electron itself has an independent occlusion observer.  My current understanding of the use of this in Chromium is that delaying this call for occlusion has no impact other than rendering the webContents for an extra few seconds